static/app/components/replays/highlightReplayPlugin.tsx

```diff
@@ -1,51 +1,83 @@
 import type {Replayer} from 'rrweb';
 import type {eventWithTime} from 'rrweb/typings/types';
 
-import type {HighlightsByTime} from 'sentry/views/replays/types';
+import {Highlight} from 'sentry/views/replays/types';
 
 interface HighlightReplayPluginOptions {
-  highlights: HighlightsByTime;
   defaultHighlightColor?: string;
 }
 
 class HighlightReplayPlugin {
-  highlights: HighlightsByTime;
   defaultHighlightColor: string;
 
   constructor({
-    highlights,
-    // TODO(replays): Figure out a better color
-    defaultHighlightColor = 'rgba(255, 192, 203, 0.5)',
-  }: HighlightReplayPluginOptions) {
+    defaultHighlightColor = 'rgba(168, 196, 236, 0.75)',
+  }: HighlightReplayPluginOptions = {}) {
     this.defaultHighlightColor = defaultHighlightColor;
-    this.highlights = highlights;
   }
 
-  handler(event: eventWithTime, _isSync: boolean, context: {replayer: Replayer}) {
-    if (!this.highlights.size) {
-      return;
-    }
-
-    const ts = Math.floor(event.timestamp / 1000);
-    const highlightObj = this.highlights.get(ts);
+  handler(event: eventWithTime, _isSync: boolean, {replayer}: {replayer: Replayer}) {
+    const highlightObj = event.data as Highlight;
 
     // ts has some issues with control flow analysis, so it will complain if we
     // use `has` to check existance. instead, ensure that it is not undefined.
     // See https://github.com/Microsoft/TypeScript/issues/9619
-    if (!highlightObj) {
+    if (highlightObj.nodeId === undefined || highlightObj?.nodeId < 0) {
```

Comment:
Inconsistent use of `?`. Looks like it's not needed per the `Highlight` type, and the first half of the cond isn't complaining

Reply:
Hmm, I guess it's a bit misleading as `event.data` is only confirmed as a `Highlight` type after this condition.